gwt-regexp/pom.xml

```diff
@@ -65,39 +67,26 @@
             <version>${elemental2.version}</version>
             <classifier>sources</classifier>
         </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>${junit.version}</version>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
     <build>
-        <resources>
-            <resource>
-                <directory>src/main/java</directory>
```

Comment:
Its been a long time, can you confirm for me that .java and .gwt.xml sources end up inside the plain jar, not just the -sources jar? I know you can technically add the sources jar, but doubling all your dependencies is silly imho.

Reply:
That said, we can discuss before just making the change, i'm open to that.

Reply:
As i mentioned before, gwt-lib with gwt plugin will actually copy the resources and the *.gwt.xml without the need to copy them manually, check below images for the result Jar:

![Screenshot from 2020-01-07 20-34-11](https://user-images.githubusercontent.com/23003164/71919340-1f176c80-318d-11ea-9b79-c337abb4a9b8.png)
![Screenshot from 2020-01-07 20-34-09](https://user-images.githubusercontent.com/23003164/71919341-1fb00300-318d-11ea-80a3-9be05c790a34.png)

Reply:
As I mentioned before, I would stay away from using the gwt-m-p.

Reply:
@FrankHossfeld unless you have evidence that this breaks either old gwt-maven-plugin or the j2cl-maven-plugin, i dont understand your objection.

Reply:
@niloc132 Well, I do not see a real benefit adding a plugin that does more or less not more than the magic of copying the resources into the jar, which could be done with a few lines of code (much clearer) inside the pom.

Reply:
@FrankHossfeld This would help with organizing the modules to be in a consistent way, i mean the module.gwt.xml is placed in one place and there is no need to worry about coping these files including the sources into the jar while the plugin does that for you. On the other hand, the module.gwt.xml is much simpler now as the plugin will add the source paths for you depends on the package you use. Also, because we are not using any GWT goals and we don't have gwt-user in our classpath, then it doesn't create any problem yet it simplifies the pom and doesn't change the usage at all.

Reply:
The gwt-m-p plugin (if I understand it correctly)

1. copys the sourcres 
2. moves the module descriptor
3. renames the module descriptor

1. can be easily done with the resource tag
2. must be configurated, In both cases you need to know where to place the module descriptor (using the plugin you have to write a cpnfiguration, without you need to create directories)
3. you need to know the name of the module descriptor.

Do I miss a benefit of the plugin?

So, using the gwt-m-p plugin
* adds complexity
* create another dependency
* adds another point of failure.

That's why I see no real benefit using the plugin. 

Regarding the organzation, I am not sure, what the benefit is haveng the module descriptor under resources compare to having it inside the resource structure.

Reply:
Here's an opposing perspective:

 * The generated jar from either plugin is _nearly_ the same - either should produce a functional jar for either plugin, or for j2cl
 * Avoiding the ltgt plugin means requiring adding a <resource> section, so that sources are correctly included, so it amounts to roughly the same amount of configuration
 * If the packaging is jar, any users of the ltgt plugin must take an extra step in order to use this project at all.

My move here would be to make the jar as easy to use for consumers, even if it adds a slight bit of complexity in the project setup, since in theory we just do that setup once and are done with it. We can also have it both ways, by skipping the ltgt module.gwt.xml generation, but still taking advantage of the "one gwt module per jar" convention that the ltgt plugin correctly enforces (and that j2cl will rely on) - this will let us move the module.gwt.xml back where it was, and still requires no <resource> stanza (packaging must still be gwt-lib for downstream projects to correctly _automatically_ add <inherits> statements).

--

With that said, you are concerned about some failure mode: do you have an example of how this can go wrong? Otherwise, the complexity appears to be roughly equal in each direction to me, and the benefits for ltgt users seem clear, with no downside for old plugin or j2cl users?

Reply:
Will we have the advantages also, in case the module uses Gradle instead of Maven?

Reply:
The ltgt plugin makes two small changes that you wouldnt otherwise have without it - it marks the packaging in the pom as `gwt-lib`, and sets up a file (in META-INF iirc) with the name of the module that should be inherited. To my knowledge, only downstream projects using ltgt's gwt-app or gwt-lib will take advantage of this - other maven projects or gradle projects (or bazel projects) will not care at all, as long as the sources are there (and for gwt2, the .gwt.xml file is there, and manually inherited as normal).

Reply:
If you think, this is a benefit, we should do it. The only thing I'll expect is, that all migraged modules behave in the same way.